From 6864a6dd28d81deba9cf55b03d4dfd9e3bbd5cb4 Mon Sep 17 00:00:00 2001
From: Gwenhael Goavec-Merou <gwenhael.goavec-merou@armadeus.com>
Date: Tue, 3 Dec 2013 12:10:15 +0100
Subject: [PATCH 120/184] ARM: dts: imx27-apf27dev: Add pinctrl for cspi,
 sdhci, leds and keys

- add chip-select pinctrl for cspi
- add card-detect for sdhci2
- add pinctrl for gpio-leds and gpio-keys

Signed-off-by: Gwenhael Goavec-Merou <gwenhael.goavec-merou@armadeus.com>
Signed-off-by: Shawn Guo <shawn.guo@linaro.org>
---
 arch/arm/boot/dts/imx27-apf27dev.dts | 34 +++++++++++++++++++++++++++++++---
 1 file changed, 31 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx27-apf27dev.dts b/arch/arm/boot/dts/imx27-apf27dev.dts
index 7c5478e..2b6d489 100644
--- a/arch/arm/boot/dts/imx27-apf27dev.dts
+++ b/arch/arm/boot/dts/imx27-apf27dev.dts
@@ -38,6 +38,8 @@
 
 	gpio-keys {
 		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio_keys>;
 
 		user-key {
 			label = "user";
@@ -48,6 +50,8 @@
 
 	leds {
 		compatible = "gpio-leds";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio_leds>;
 
 		user {
 			label = "Heartbeat";
@@ -61,7 +65,7 @@
 	fsl,spi-num-chipselects = <1>;
 	cs-gpios = <&gpio4 28 GPIO_ACTIVE_LOW>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_cspi1>;
+	pinctrl-0 = <&pinctrl_cspi1 &pinctrl_cspi1_cs>;
 	status = "okay";
 };
 
@@ -71,7 +75,7 @@
 		   <&gpio4 27 GPIO_ACTIVE_LOW>,
 		   <&gpio2 17 GPIO_ACTIVE_LOW>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_cspi2>;
+	pinctrl-0 = <&pinctrl_cspi2 &pinctrl_cspi2_cs>;
 	status = "okay";
 };
 
@@ -111,6 +115,10 @@
 			>;
 		};
 
+		pinctrl_cspi1_cs: cspi1csgrp {
+			fsl,pins = <MX27_PAD_CSPI1_SS0__GPIO4_28 0x0>;
+		};
+
 		pinctrl_cspi2: cspi2grp {
 			fsl,pins = <
 				MX27_PAD_CSPI2_MISO__CSPI2_MISO 0x0
@@ -119,6 +127,22 @@
 			>;
 		};
 
+		pinctrl_cspi2_cs: cspi2csgrp {
+			fsl,pins = <
+				MX27_PAD_CSI_D5__GPIO2_17 0x0
+				MX27_PAD_CSPI2_SS0__GPIO4_21 0x0
+				MX27_PAD_CSPI1_SS1__GPIO4_27 0x0
+			>;
+		};
+
+		pinctrl_gpio_leds: gpioledsgrp {
+			fsl,pins = <MX27_PAD_PC_VS1__GPIO6_14 0x0>;
+		};
+
+		pinctrl_gpio_keys: gpiokeysgrp {
+			fsl,pins = <MX27_PAD_PC_VS2__GPIO6_13 0x0>;
+		};
+
 		pinctrl_imxfb1: imxfbgrp {
 			fsl,pins = <
 				MX27_PAD_CLS__CLS 0x0
@@ -181,6 +205,10 @@
 				MX27_PAD_SD2_D3__SD2_D3 0x0
 			>;
 		};
+
+		pinctrl_sdhc2_cd: sdhc2cdgrp {
+			fsl,pins = <MX27_PAD_TOUT__GPIO3_14 0x0>;
+		};
 	};
 };
 
@@ -188,7 +216,7 @@
 	bus-width = <4>;
 	cd-gpios = <&gpio3 14 GPIO_ACTIVE_HIGH>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sdhc2>;
+	pinctrl-0 = <&pinctrl_sdhc2 &pinctrl_sdhc2_cd>;
 	status = "okay";
 };
 
-- 
1.8.5.3

